libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/native/KotlinNativeTargetConfigurator.kt

```diff
@@ -483,6 +483,14 @@ open class KotlinNativeTargetConfigurator<T : KotlinNativeTarget> : AbstractKotl
                         .configurations
                         .getByName(compilation.pluginConfigurationName)
                 }
+
+                (compilation.target as? KotlinNativeTarget)?.binaries?.forEach { binary ->
+                    binary.linkTaskProvider.configure {
+                        it.compilerPluginClasspath = project
+                            .configurations
+                            .getByName(compilation.pluginConfigurationName)
+                    }
+                }
```

Comment:
Applying compiler plugins to Native "link" tasks seems risky -- the consequences are not clear and generally hardly predictable. I mean, this potentially can break existing projects, because they will unexpectedly get existing plugins applied to the link tasks.
 
Also, this approach might imply additional maintenance overhead. In particular, does it support the new binaries DSL (https://github.com/JetBrains/kotlin/blob/3085f6bacf6e1e73ac8e0812312eca70f39da929/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/native/tasks/artifact/KotlinNativeLinkArtifactTask.kt#L26)?

The question is: have you considered any alternatives?